pinot-common/src/main/java/com/linkedin/pinot/common/data/DateTimeFieldSpec.java

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.data;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.apache.commons.lang3.EnumUtils;
+import org.codehaus.jackson.annotate.JsonIgnore;
+import org.codehaus.jackson.annotate.JsonIgnoreProperties;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.linkedin.pinot.common.utils.EqualityUtils;
+
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public final class DateTimeFieldSpec extends FieldSpec {
+
+  private static final String FORMAT_TOKENS_ERROR_STR = "format must be of pattern size:timeunit:timeformat(:pattern)";
+  private static final String FORMAT_PATTERN_ERROR_STR = "format must be of format [0-9]+:<TimeUnit>:<TimeFormat>(:pattern)";
+  private static final String TIME_FORMAT_ERROR_STR =
+      "format must be of format [0-9]+:<TimeUnit>:EPOCH or [0-9]+:<TimeUnit>:SIMPLE_DATE_FORMAT:<format>";
+  private static final String GRANULARITY_TOKENS_ERROR_STR = "granularity must be of format size:timeunit";
+  private static final String GRANULARITY_PATTERN_ERROR_STR = "granularity must be of format [0-9]+:<TimeUnit>";
+  private static final String NUMBER_REGEX = "[1-9][0-9]*";
+  private static final String COLON_SEPARATOR = ":";
+
+  private String _format;
+  private String _granularity;
+  private DateTimeType _dateTimeType;
+
+  public enum DateTimeType {
```

Comment:
I would like to have a method in Pinot that takes DateTimeSpec and a value and returns a DateTime object. This can be util or part of this class

Reply:
Will add this in another PR